### PR TITLE
docstring fixes

### LINKED ISSUE #371: Improve Quickstart
We need to improve official [Quickstart](https://pytorch.org/ignite/quickstart.html) documentation:

- Empty link on ignite modules that are empty, for example: 
  - create_supervised_trainer()
  - create_supervised_evaluator()
  - Engine
  - add_event_handler()
  - on()
- Typos

### PR DESCRIPTION
**Pull request by anmolsjoshi**
Fixes #371 

Description:
* Updates empty links in quickstart.rst
* Formatting
* Small typo

Check list:
* [ ] New tests are added (if a new feature is modified)
* [x] New doc strings: text and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@anmolsjoshi could you please render online the docs (e.g. at your anmolsjoshi.github.io/ignite/) to check your modifications. I built locally the docs and some fixes still needed. For example,

screen shot 2019-01-09 at 08 54 55

**Comment by vfdev-5** [comment]
> Another thing I caught was Regression metrics are showing compute and reset of Metric.

@anmolsjoshi probably these methods are not documented in regression classes. IMO, this is OK.

**Comment by anmolsjoshi** [comment]
Works for me!

File: ignite/engine/__init__.py [hunk 1]
```diff
@@ -53,20 +53,20 @@ def create_supervised_evaluator(model, metrics={},
...
             Applies to both model and batches.
         non_blocking (bool, optional): if True and this copy is between CPU and GPU, the copy may occur asynchronously
             with respect to the host. For other cases, this argument has no effect.
-        prepare_batch (Callable, optional): function that receives `batch`, `device`, `non_blocking` and outputs
+        prepare_batch (callable, optional): function that receives `batch`, `device`, `non_blocking` and outputs
             tuple of tensors `(batch_x, batch_y)`.
 
     Returns:
-        Engine: an evaluator engine with supervised inference function
+        :class:`~ignite.engine.Engine`: an evaluator engine with supervised inference function.
```
Comment (vfdev-5) [inline ignite/engine/__init__.py]: IMO, it was OK
```suggestion
        Engine: an evaluator engine with supervised inference function.
```

File: ignite/engine/__init__.py [hunk 2]
```diff
@@ -17,21 +17,21 @@ def create_supervised_trainer(model, optimizer, loss_fn,
...
             Applies to both model and batches.
         non_blocking (bool, optional): if True and this copy is between CPU and GPU, the copy may occur asynchronously
             with respect to the host. For other cases, this argument has no effect.
-        prepare_batch (Callable, optional): function that receives `batch`, `device`, `non_blocking` and outputs
+        prepare_batch (callable, optional): function that receives `batch`, `device`, `non_blocking` and outputs
             tuple of tensors `(batch_x, batch_y)`.
 
     Returns:
-        Engine: a trainer engine with supervised update function
+        :class:`~ignite.engine.Engine`: a trainer engine with supervised update function.
```
Comment (vfdev-5) [inline ignite/engine/__init__.py]:
```suggestion
        Engine: a trainer engine with supervised update function.
```

File: ignite/engine/engine.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 
 class Events(Enum):
-    """Events that are fired by the :class:`ignite.engine.Engine` during execution"""
+    """Events that are fired by the :class:`Engine` during execution."""
```
Comment (vfdev-5) [inline ignite/engine/engine.py]:
```suggestion
    """Events that are fired by the :class:`~ignite.engine.Engine` during execution."""
```

File: ignite/handlers/checkpoint.py [hunk 4]
```diff
@@ -39,14 +39,14 @@ class ModelCheckpoint(object):
...
-            in the directory 'dirname'
+            in the directory 'dirname'.
         create_dir (bool, optional):
             If True, will create directory 'dirname' if it doesnt exist.
         save_as_state_dict (bool, optional):
             If True, will save only the `state_dict` of the objects specified, otherwise the whole object will be saved.
 
     Notes:
-          This handler expects two arguments: an `Engine` object and a `dict`
+          This handler expects two arguments: an :meth:`~ignite.engine.Engine` object and a `dict`
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py]:
```suggestion
          This handler expects two arguments: an :class:`~ignite.engine.Engine` object and a `dict`
```

File: ignite/handlers/early_stopping.py [hunk 5]
```diff
@@ -4,16 +4,16 @@
...
-            Number of events to wait if no improvement and then stop the training
-        score_function (Callable):
-            It should be a function taking a single argument, an `ignite.engine.Engine` object,
+            Number of events to wait if no improvement and then stop the training.
+        score_function (callable):
+            It should be a function taking a single argument, an :class:`ignite.engine.Engine` object,
             and return a score `float`. An improvement is considered if the score is higher.
-        trainer (Engine):
-            trainer engine to stop the run if no improvement
+        trainer (ignite.engine.Engine):
```
Comment (vfdev-5) [inline ignite/handlers/early_stopping.py]:
```suggestion
        trainer (Engine):
```

**Comment by vfdev-5** [comment]
@anmolsjoshi what is the difference between 
```
:class:`~ignite.engine.Engine`
```
and 
```
:class:`ignite.engine.Engine`
```
?

File: ignite/metrics/running_average.py [hunk 6]
```diff
@@ -9,8 +9,10 @@ class RunningAverage(Metric):
         src (Metric or None): input source: an instance of :class:`ignite.metrics.Metric` or None. The latter
             corresponds to `engine.state.output` which holds the output of process function.
         alpha (float, optional): running average decay factor, default 0.98
-        output_transform (Callable, optional): a function to use to transform the output if `src` is None and
-            corresponds the output of process function. Otherwise it should be None.
+        output_transform (callable, optional): a callable that is used to transform the
```
Comment (vfdev-5) [inline ignite/metrics/running_average.py]: What ???

**Comment by anmolsjoshi** [comment]
:class:ignite:engine.Engine results in a the text `ignite.engine.Engine`

:class:~ignite:engine.Engine results in a the text `Engine`

**Comment by vfdev-5** [comment]
Okay, let's take the convention to use only 
```
:class:~ignite.some_module.ClassName
```
for ignite's classes and methods

File: ignite/metrics/running_average.py (hunk 6 above)
Comment (vfdev-5) [reply ignite/metrics/running_average.py]: @anmolsjoshi in case of `RunningAverage`, the role of `output_transform` is the same as for `Metric` and in addition there is a condition on when it should be defined and when it should be None depending on `src`.

File: ignite/engine/engine.py [hunk 7]
```diff
@@ -105,20 +105,20 @@ class Custom_Events(Enum):
...
-            *args: optional args to be passed to `handler`
-            **kwargs: optional keyword args to be passed to `handler`
+                :class:`Events` or any `event_name` added by :meth:`register_events`.
+            handler (callable): the callable event handler that should be invoked
+            *args: optional args to be passed to `handler`.
+            **kwargs: optional keyword args to be passed to `handler`.
 
         Notes:
-              The handler function's first argument will be `self`, the `Engine` object it was bound to.
+              The handler function's first argument will be `self`, the :class:`Engine` object it was bound to.
```
Comment (vfdev-5) [inline ignite/engine/engine.py]:
```suggestion
              The handler function's first argument will be `self`, the :class:`~ignite.engine.Engine` object it was bound to.
```

**Comment by anmolsjoshi** [comment]
@vfdev-5 all comments have been incorporated. please let me know if anything else is needed.
